master/docs/manual/configuration/workers-docker.rst

```diff
@@ -469,10 +469,34 @@ With :class:`KubeHardcodedConfig`, you just configure the necessary parameters t
 
 ``headers``
     (optional)
-    The headers necessary for authentication if needed.
-    For example, for BasicAuth, you would say::
+    Additional headers to be passed to the HTTP request
 
-        headers={"Authorization": "Basic %s" % base64.b64encode("userid:password")}
+``authorization``
+    (optional)
+    Authorization header, for instance:
+
+        authorization="Basic %s" % base64.b64encode("userid:password")
+
+    Unlike the headers argument, this argument supports secret providers, so
+    you could store e.g. a bearer token in a secret provider and do
+
+        authorization=util.Interpolate("Bearer %(secret:k8s-token)")
+
+    When using the Google Kubernetes Engine (GKE), a bearer token for the
+    default service account can be had with:
+
+        gcloud container clusters get-credentials --region [YOURREGION] YOURCLUSTER
+        kubectl describe sa
+        kubectl describe secret [SECRET_ID]
```

Comment:
My understanding is that this secret is temporary, and will only last for a limited time. kubectl is supposed to automatically renew that token.

So whereas this method might work, if I understand correctly, it is not suitable for prod.

Reply:
Well ... I'm running with that, I'll report my findings by the next week ;-)

My understanding - but it is very unclear in the docs - is that the token for the user has a limited time, not the token for a service account. `get-credentials` recommends using service accounts for external applications that need to access the Kubernetes cluster.

Reply:
Why don't you use `KubeProxyConfigLoader`? It was written especially for the weird renewable token of GKE.
That is what we are using for metabuildbot.

Reply:
I'm running the master on-premise, only the slaves on GKE. Or have I misunderstood how KubeProxyConfigLoader works ?

Reply:
OK ... not right. KubeCtlProxyConfigLoader is using kubectl. I've added a commit to describe the three options at a high level.

I find it a lot more convenient than having to build a container with gcloud setup and logged in, as well as with kubectl installed, to be honest. I agree that it would be a solution, but for what it's worth the bearer token seems to work so far.

I can't find any documentation on automatic rotation of tokens on GKE. Do you have a link ?

Reply:
I have implied this token rotation from that github issue: https://github.com/LeastAuthority/txkube/issues/139

Reply:
I agree that gcloud creds provisioning might not be convenient. FWIW This is what I've done in our ansible setup: https://github.com/buildbot/buildbot-infra/blob/master/roles/bb-master/tasks/main.yml#L141